genome_preparation.nf

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env nextflow
+
+def helpMessage() {
+    log.info"""
+    Index the provided genome with bowtie and samtools
+    Usage:
+            --fasta                     Fasta file containing the reference genome.
+            --outdir                    Directory where th index will be deployed.
+    """
+}
+
+// Show help message
+if (params.help) exit 0, helpMessage()
+
+params.fasta  = "$baseDir/resources/genome.fa"
+params.outdir = "$baseDir/resources/"
+
+/*
+ * Index the provided reference genome using bowtie build
+ */
+process bowtieGenomeIndex {
+
+    publishDir params.outdir,
+        overwrite: false,
+        mode: "move",
+        saveAs: { filename -> file(params.fasta).getName() + filename.replaceFirst(/reference/, "") }
+
+    input:
+        path 'reference' from params.fasta
+
+    output:
+        path "reference.*.bt2"
+
+    """
+    bowtie2-build reference reference
```

Comment:
actually, this command puzzles me. are those "reference reference" files? directories? it looks like they are not variables from the nextflow process...

Reply:
I agree this is a confusing command. See in the [remapping commands](https://github.com/EBIvariation/variant-remapping/blob/master/remapping_commands.sh#L105). 
`reference` here is simply a string for a relative path. Nextflow has copied the input genome  to a local directory following the `input` directive and will move the output to its final location following the `publishDir` directive.
I should change the second string to index to make it more obvious which is which.